planner/core/physical_plans.go

```diff
@@ -108,23 +108,16 @@ func (p *PhysicalTableReader) GetTablePlan() PhysicalPlan {
 	return p.tablePlan
 }
 
-// GetTableScan exports the tableScan that contained in tablePlan.
-func (p *PhysicalTableReader) GetTableScan() *PhysicalTableScan {
-	curPlan := p.tablePlan
-	for {
-		chCnt := len(curPlan.Children())
-		if chCnt == 0 {
-			return curPlan.(*PhysicalTableScan)
-		} else if chCnt == 1 {
-			curPlan = curPlan.Children()[0]
-		} else {
-			join, ok := curPlan.(*PhysicalHashJoin)
-			if !ok {
-				return nil
-			}
-			curPlan = join.children[1-join.globalChildIndex]
+// GetTableScans exports the tableScan that contained in tablePlan.
```

Comment:
I suggest to keep both `GetTableScan` and `GetTableScans`, and throw error if there are more than 1 tables when calling `GetTableScan`